### PR TITLE
Fancy fit functions for resonators.

### PR DESCRIPTION
**Pull request by gribeill**
Add fit functions for biased Lorentzian, and hanger resonator in complex plane that compensates for phase delay and scaling/rotation. 
My first PR to Qlab, so do let me know about any style issues...
-GJR

### CODE REVIEW

**Comment by caryan** [comment]
@gribeill look pretty good.  The only thing I can spot is hard coded figure numbers.

**Comment by dieris** [comment]
@gribeill you can look at the end of this PR for reference: https://github.com/BBN-Q/Qlab/pull/41

**Comment by gribeill** [comment]
Done. Thanks for pointing me to an example @dieris.

File: analysis/Resonator_fitting/fit_biased_lorentzian.m [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        figure();
+        hold on;
+        plot(freq, data, '.', 'MarkerSize', 15);
+        plot(freq, ffit, 'r-', 'LineWidth', 2);
+        plot(freq, ffit+1.96*pband, 'r--', 'LineWidth', 2);
+        plot(freq, ffit-1.96*pband, 'r--', 'LineWidth', 2);
+        grid on;
+        xlabel('Frequency', 'FontSize', 15);
+        ylabel('Amplitude', 'FontSize', 15);
+        title(sprintf('Biased Lorentzian Fit, f_0 = %.3f, FWHM = %.3f', fit_params(2), 2*fit_params(3)), 'FontSize', 12);
```
Comment (dieris) [inline analysis/Resonator_fitting/fit_biased_lorentzian.m]: @gribeill Here you print 2*params(3) as FWHM, but below you plot params(3) (see other comment). What do you expect as data, amplitude or power?

File: analysis/Resonator_fitting/fit_biased_lorentzian.m [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        xlabel('Frequency', 'FontSize', 15);
+        title('Fit', 'FontSize', 12);
+        set(gca, 'FontSize', 14);
+        
+        figure();
+        subplot(2,1,1);
+        errorbar(1:M, fit_params(:,2), 1.96*sqrt(fit_vars(:,2)), '.', 'MarkerSize', 15, 'LineWidth', 2);
+        ylabel('f_0', 'FontSize', 15); set(gca, 'FontSize', 14); grid on;
+        subplot(2,1,2);
+        errorbar(1:M, fit_params(:,3), 1.96*sqrt(fit_vars(:,3)), '.', 'MarkerSize', 15, 'LineWidth', 2);
```
Comment (dieris) [inline analysis/Resonator_fitting/fit_biased_lorentzian.m]: here FWHM = params(3). Which one? I believe that it should be 2*params(3) if the data is in amplitudes

**Comment by blakejohnson** [comment]
With these new additions, can we clear out a bunch of the old cruft in the `analysis/Resonator_fitting` folder? I see a bunch of things in there that depend on a data format we haven't used in 5 years, along with older, less robust circle fitting routines.

File: analysis/Resonator_fitting/fit_biased_lorentzian.m [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+options = optimset('MaxIter', parser.Results.MaxIter);
+
+fit_params = zeros(M, 5);
+fit_vars = zeros(M, 5);
+ssemin = zeros(M,1);
+
+for idx = 1:M  
+    p0 = initial_guess(freq, data(:,idx));
+    model = @(p)bias_lorentz_model(p, freq, data(:,idx));
+    [p, ssemin(idx)] = fminsearch(model, p0, options);
```
Comment (blakejohnson) [inline analysis/Resonator_fitting/fit_biased_lorentzian.m]: why `fminsearch` instead of `nlinfit`?

**Comment by blakejohnson** [comment]
I'd love to see Julia versions of these methods in Qlab.jl.

File: analysis/Resonator_fitting/fit_biased_lorentzian.m (hunk 3 above)
Comment (gribeill) [inline analysis/Resonator_fitting/fit_biased_lorentzian.m]: Matt's guess is right.  `fminsearch` does okay even if it's not ideal, although someone should probably double check that I'm calculating confidence intervals correctly.

**Comment by gribeill** [comment]
@blakejohnson I'd be happy to Julia-ify these methods.

**Pull request merged**